File: module_me/src/main/java/com/fly/tour/me/contract/NewsTypeAddContract.java
package com.fly.tour.me.contract;

import com.fly.tour.api.dto.RespDTO;
import com.fly.tour.api.newstype.entity.NewsType;
import com.fly.tour.common.mvp.BaseView;

import io.reactivex.Observable;

/**
 * Description: <NewsTypeAddContract><br>
 * Author:      mxdl<br>
 * Date:        2019/5/24<br>
 * Version:     V1.0.0<br>
 * Update:     <br>
 */
public interface NewsTypeAddContract {
    interface Presenter{
        void addNewsType(String typename);
    }
    interface View extends BaseView {

    }
    interface Model{
        Observable<RespDTO<NewsType>> addNewsType(NewsType type);
    }
}

File: plugin_fly_me/src/main/java/com/fly/tour/me/contract/NewsTypeAddContract.kt
package com.fly.tour.me.contract

import com.fly.tour.common.mvp.BaseView

/**
 * Description: <NewsTypeAddContract><br>
 * Author:      mxdl<br>
 * Date:        2020/2/17<br>
 * Version:     V1.0.0<br>
 * Update:     <br>
 */
interface NewsTypeAddContract {
    interface Presenter{
        fun addNewsType(typename: String)
    }
    interface View: BaseView{

    }
    interface Model{
        fun addNewsType(type: String): Boolean
    }
}